apps/wallet/src/ui/app/pages/qredo-connect/utils.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type UIQredoPendingRequest } from '_src/background/qredo/types';
+
+export function isUntrustedQredoConnect({
+    apiUrl,
+    origin,
+}: UIQredoPendingRequest) {
+    try {
+        return (
+            new URL(origin).protocol !== 'https:' ||
+            new URL(apiUrl).protocol !== 'https:'
+        );
+    } catch (e) {
+        return false;
```

Comment:
if the URLs aren't valid, should we say they're untrusted?

Reply:
Yeah that's fine I think, especially in this case. We already do a validation for the url